tasks/release.py

```diff
@@ -1022,3 +897,396 @@ def tag_version(ctx, agent_version, commit="HEAD", verify=True, push=True, force
+            # - if the previous tag is a devel tag, use it without the devel tag
+            # - otherwise (should not happen during regular release cycles), bump the minor version
+            if previous_version.is_devel():
+                new_version = previous_version.non_devel_version()
+                new_version = new_version.next_version(rc=True)
+            else:
+                new_version = previous_version.next_version(bump_minor=True, rc=True)
+
+    return new_version
+
+
+def fail_if_not_base_branch(branch, release_version):
+    """
+    Checks if the given branch is either the default branch or the release branch associated
+    with the given release version.
+    """
+    if branch != DEFAULT_BRANCH and branch != "{}.{}.x".format(release_version.major, release_version.minor):
+        raise Exit(
+            color_message(
+                "The branch you are on is neither {} or the correct release branch ({}.{}.x). Aborting.".format(
+                    DEFAULT_BRANCH, release_version.major, release_version.minor
+                ),
+                "red",
+            ),
+            code=1,
+        )
+
+
+def check_uncommitted_changes(ctx):
+    """
+    Checks if there are uncommitted changes in the local git repository.
+    """
+    modified_files = ctx.run("git --no-pager diff --name-only HEAD | wc -l", hide=True).stdout.strip()
+
+    # Return True if at least one file has uncommitted changes.
+    return modified_files != "0"
+
+
+def check_local_branch(ctx, branch):
+    """
+    Checks if the given branch exists locally
+    """
+    matching_branch = ctx.run("git --no-pager branch --list {} | wc -l".format(branch), hide=True).stdout.strip()
+
+    # Return True if a branch is returned by git branch --list
+    return matching_branch != "0"
+
+
+def check_upstream_branch(github, branch):
+    """
+    Checks if the given branch already exists in the upstream repository
+    """
+    github_branch = github.get_branch(branch)
+
+    # Return True if the branch exists
+    return github_branch and github_branch.get('name', False)
+
+
+def parse_major_versions(major_versions):
+    return sorted([int(x) for x in major_versions.split(",")])
+
+
+@task
+def finish(ctx, major_versions="6,7"):
+    """
+    Updates the release entry in the release.json file for the new version.
+
+    Updates internal module dependencies with the new version.
+    """
+
+    if sys.version_info[0] < 3:
+        print("Must use Python 3 for this task")
+        return Exit(code=1)
+
+    list_major_versions = parse_major_versions(major_versions)
+    print("Finishing release for major version(s) {}".format(list_major_versions))
+
+    github_token = get_github_token()
+
+    for major_version in list_major_versions:
+        new_version = next_final_version(ctx, major_version)
+        update_release_json(github_token, new_version)
+
+    # Update internal module dependencies
+    update_modules(ctx, str(new_version))
+
+
+@task
```

Comment:
Remove `upstream` sentence from docs and add here
```suggestion
@task(help={'upstream': "Remote repository name (default 'origin')"})
```